src/OrchardCore/OrchardCore/Shell/Configuration/ShellConfigurationSources.cs

```diff
@@ -70,5 +70,20 @@ public async Task SaveAsync(string tenant, IDictionary<string, string> data)
                 }
             }
         }
+
+        public Task DeleteAsync(string tenant)
+        {
+            lock (this)
```

Comment:
A small think but locking on `this` is not the best. Also `ShellsSettingsSources.cs`

Reply:
As i remember, for other methods, this is the caller (shell settings manager) that uses async semaphores.